docs/linux.md

```diff
@@ -25,58 +25,82 @@ sudo yum -y install gcc gcc-c++ python3-setuptools git-clang-format dos2unix rub
 ```
 ## 2. Packages Requirements when Executing Binaries:
 
-OpenSSL* toolkit version 1.1.1q
+OpenSSL* toolkit version 3.0.5.
 GCC version > 7.5
-#### Steps to Upgrade the OpenSSL* Toolkit to Version 1.1.1q
 
-1. Pull the tarball:
-	```
-	wget https://www.openssl.org/source/openssl-1.1.1q.tar.gz
+#### Steps to Upgrade the OpenSSL* Toolkit to Version 3.0.5
+
+You can either perform the steps in this section manually or use the install_openssl_curl.sh script from the source folder. Run this script with "sudo bash install_openssl_curl.sh -u 1.1.1q" to remove manually installed openssl 1.1.1 version. Then "sudo bash install_openssl_curl.sh -i 3.0.5" to install openssl 3 and curl upgraded with openssl 3.
+
+1. If libssl-dev, curl and libcurl are already installed, remove it:
 	```
-2. Unpack the tarball with:
+	sudo apt-get remove --auto-remove libssl-dev
+	sudo apt-get remove --auto-remove libssl-dev:i386
+	sudo apt remove curl libcurl4-openssl-dev
 	```
-	tar -zxf openssl-1.1.1q.tar.gz && cd openssl-1.1.1q
+    In case of RHEL OS, use below commands to uninstall:
 	```
-3. Issue the command:
+	yum remove curl libcurl-devel openssl-devel
 	```
+2. In case if openssl 1.1.1 version is manually installed on the system before, uninstall it.
+   Can skip this step if the system has only the default openssl that comes along with OS.
+    ```
+	wget https://www.openssl.org/source/openssl-1.1.1q.tar.gz
+	tar -xvzf openssl-1.1.1q.tar.gz && cd openssl-1.1.1q
 	./config
+	make -j$(nproc)
+	sudo make uninstall
+	rm /usr/bin/openssl
+	rm -rf openssl-1.1.1q
+	rm -f openssl-1.1.1q.tar.gz
+	ldconfig
+	cd ..
 	```
-4. Issue the command:
-	```
-	make
-	```
-	  (You may need to run `sudo apt install make gcc` before running this command successfully).
 
-5. Check for possible errors:
-	```
-	make test
-	```
-6. Backup the current OpenSSL* binary:
+3. Install openssl 3 manually with below steps:
 	```
+	#Pull the tarball
+	wget https://www.openssl.org/source/openssl-3.0.5.tar.gz
+	#Unpack the tarball
+	tar -zxf openssl-3.0.5.tar.gz && cd openssl-3.0.5
+	./config
+	make -j$(nproc)
+	#Backup the current OpenSSL* binary
 	sudo mv /usr/bin/openssl ~/tmp
-	```
-7. Issue the command:
-	```
 	sudo make install
+	#Create a symbolic link from the newly installed binary to the default location
+	sudo ln -s /usr/local/bin/openssl /usr/bin/openssl
+	#Run the command to update symlinks and rebuild the library cache
+	cat /usr/local/lib64/ >> /etc/ld.so.conf.d/libc.conf
```

Comment:
this is not needed since "grep -qxF '/usr/local/lib64/' /etc/ld.so.conf.d/libc.conf || echo /usr/local/lib64/ | sudo tee -a /etc/ld.so.conf.d/libc.conf" is doing same